File: frontend/types/education.ts
export interface EducationCourse {
    ID: number;
    title: string;
    description: string;
    organization: string;
    image_url?: string;
    is_published: boolean;
    scripture_book_id?: number;
    modules?: EducationModule[];
}

export interface EducationModule {
    ID: number;
    course_id: number;
    title: string;
    description: string;
    order: number;
    questions?: ExamQuestion[];
}

export interface ExamQuestion {
    ID: number;
    module_id: number;
    text: string;
    type: 'multiple_choice' | 'boolean';
    organization?: string;
    verse_reference?: string;
    points: number;
    options: AnswerOption[];
}

export interface AnswerOption {
    ID: number;
    question_id: number;
    text: string;
    is_correct?: boolean; // Usually hidden from client, but returned in submit result
    explanation?: string;
}

export interface UserExamAttempt {
    ID: number;
    user_id: number;
    module_id: number;
    score: number;
    total_points: number;
    passed: boolean;
    completed_at: string;
}

export interface UserModuleProgress {
    user_id: number;
    module_id: number;
    status: 'not_started' | 'in_progress' | 'completed';
    progress_percent: number;
    last_accessed_at: string;
}

export type TutorHistoryRole = 'user' | 'assistant';

export interface TutorHistoryMessage {
    role: TutorHistoryRole;
    content: string;
}

export interface TutorSource {
    id: string;
    domain: string;
    sourceType: string;
    sourceId: string;
    title: string;
    snippet: string;
    sourceUrl?: string;
    score?: number;
    metadata?: Record<string, unknown>;
}

export interface TutorAssistantContext {
    domains: string[];
    sources: TutorSource[];
    confidence: number;
    language?: string;
    visibility_scope?: string;
}

export interface TutorWeakTopic {
    topicKey: string;
    topicLabel: string;
    mastery: number;
    lastSeenAt: string;
    source: string;
}

export interface TutorModelInfo {
    selected: string;
    route: string;
    fallbackUsed: boolean;
}

export interface TutorTurnRequest {
    message: string;
    history?: TutorHistoryMessage[];
    topK?: number;
    forceReasoning?: boolean;
}

export interface TutorTurnResponse {
    reply: string;
    assistant_context: TutorAssistantContext;
    weak_topics: TutorWeakTopic[];
    model: TutorModelInfo;
}

export interface TutorWeakTopicsResponse {
    items: TutorWeakTopic[];
}

export type TutorMemoryScope = 'all' | 'memory' | 'weak_topics';

export interface TutorMemoryClearResponse {
    deleted: {
        memoryDocs: number;
        weakTopicDocs: number;
        weakTopics: number;
    };
}

export interface TutorStatusResponse {
    enabled: boolean;
}

File: server/internal/models/education.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type EducationCourse struct {
	gorm.Model
	Title           string        `json:"title"`
	Description     string        `gorm:"type:text" json:"description"`
	Organization    string        `gorm:"index" json:"organization"` // e.g., "ISKCON", "Gaudiya Math"
	ImageURL        string        `json:"image_url"`
	IsPublished     bool          `gorm:"default:false" json:"is_published"`
	ScriptureBookID *uint         `json:"scripture_book_id"`
	ScriptureBook   *ScriptureBook `gorm:"foreignKey:ScriptureBookID" json:"scripture_book,omitempty"`
	Modules         []EducationModule `gorm:"foreignKey:CourseID" json:"modules,omitempty"`
}

type EducationModule struct {
	gorm.Model
	CourseID    uint            `gorm:"index" json:"course_id"`
	Title       string          `json:"title"`
	Description string          `gorm:"type:text" json:"description"`
	Order       int             `json:"order"`
	Questions   []ExamQuestion  `gorm:"foreignKey:ModuleID" json:"questions,omitempty"`
}

type ExamQuestion struct {
	gorm.Model
	ModuleID       uint           `gorm:"index" json:"module_id"`
	Text           string         `gorm:"type:text" json:"text"`
	Type           string         `gorm:"default:'multiple_choice'" json:"type"` // multiple_choice, boolean
	Organization   string         `gorm:"index" json:"organization"`             // To override course org if needed
	VerseReference string         `json:"verse_reference"`                       // Optional link to ScriptureVerse code
	Points         int            `gorm:"default:1" json:"points"`
	Options        []AnswerOption `gorm:"foreignKey:QuestionID" json:"options,omitempty"`
}

type AnswerOption struct {
	gorm.Model
	QuestionID  uint   `gorm:"index" json:"question_id"`
	Text        string `json:"text"`
	IsCorrect   bool   `json:"is_correct"`
	Explanation string `gorm:"type:text" json:"explanation"`
}

type UserExamAttempt struct {
	gorm.Model
	UserID      uint      `gorm:"index" json:"user_id"`
	User        User      `gorm:"foreignKey:UserID" json:"user,omitempty"`
	ModuleID    uint      `gorm:"index" json:"module_id"`
	Module      EducationModule `gorm:"foreignKey:ModuleID" json:"module,omitempty"`
	Score       int       `json:"score"`
	TotalPoints int       `json:"total_points"`
	Passed      bool      `json:"passed"`
	CompletedAt time.Time `json:"completed_at"`
}

type UserModuleProgress struct {
	gorm.Model
	UserID          uint            `gorm:"index" json:"user_id"`
	ModuleID        uint            `gorm:"index" json:"module_id"`
	Status          string          `gorm:"default:'not_started'" json:"status"` // not_started, in_progress, completed
	ProgressPercent int             `gorm:"default:0" json:"progress_percent"`
	LastAccessedAt  time.Time       `json:"last_accessed_at"`
}
